cmake/macros/macro_deal_ii_invoke_autopilot.cmake

```diff
@@ -46,14 +46,38 @@ macro(deal_ii_invoke_autopilot)
 
   set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
 
-  # Define and setup a compilation target:
+  #
+  # Make sure that compile_commands.json is available in the source
+  # directory:
+  #
+
+  # In order to check whether we have an "in source" build we have to make
+  # sure to remove all symlinks and resolve to absolute file paths. This is
+  # necessary because with symlinks CMAKE_SOURCE_DIR and CMAKE_BINARY dir
+  # might look different but are in fact the same (fully resolved)
+  # directory...
+  get_filename_component(_source "${CMAKE_SOURCE_DIR}" REALPATH)
+  get_filename_component(_build  "${CMAKE_BINARY_DIR}" REALPATH)
+  if(NOT "${_source}" STREQUAL "${_build}")
```

Comment:
I am not sure why this check is necessary as the file would be there if it is a build directory anyways.

Reply:
Because the `compile_commands.json` will be created after the configure phase when cmake knows about all targets. Meaning at the time when we execute this part of our `invoke_autopilot` macro the file might not have been written yet.